smtpserver/server.go

```diff
@@ -2418,10 +2418,16 @@ func (c *conn) deliver(ctx context.Context, recvHdrFor func(string) string, msgW
 					m.MessageID = messageid
 					m.MessageHash = messagehash
 					acc.WithWLock(func() {
-						if hasSpace, err := acc.TidyRejectsMailbox(c.log, conf.RejectsMailbox); err != nil {
+						hasSpace := true
+						var err error = nil
+						if !conf.KeepRejects {
+							hasSpace, err = acc.TidyRejectsMailbox(c.log, conf.RejectsMailbox)
+						}
+						if err != nil {
 							log.Errorx("tidying rejects mailbox", err)
-						} else if hasSpace {
-							if err := acc.DeliverMailbox(log, conf.RejectsMailbox, m, dataFile, false); err != nil {
+						}
+						if hasSpace {
+							if err = acc.DeliverMailbox(log, conf.RejectsMailbox, m, dataFile, false); err != nil {
```

Comment:
in mox i keep the shadowing for err (not other vars). or is there another reason for this change besides reducing shadowing?

```suggestion
							if err := acc.DeliverMailbox(log, conf.RejectsMailbox, m, dataFile, false); err != nil {
```

Reply:
I just have no idea what I'm doing with Go. Done!